### PR TITLE
style: update the use of LegacyKeyValueFormat in Dockerfiles

### LINKED ISSUE #486: style: update the use of LegacyKeyValueFormat in Dockerfiles
Per Docker [build-checks](https://docs.docker.com/reference/build-checks/legacy-key-value-format/):

> The correct format for declaring environment variables and build arguments in a Dockerfile is `ENV key=value` and `ARG key=value`, where the variable name (`key`) and value (`value`) are separated by an equals sign (`=`). Historically, Dockerfiles have also supported a space separator between the key and the value (for example, `ARG key value`). This legacy format is deprecated, and you should only use the format with the equals sign.

This admonition does not mention `LABEL`, but it is included. 

Update the use of `LABEL` and `ENV` where necessary to remove deprecation warning.

### PR DESCRIPTION
**Pull request by timothystone**
Per Docker [build-checks](https://docs.docker.com/reference/build-checks/legacy-key-value-format/):

  The correct format for declaring environment variables and build arguments in a Dockerfile is
  `ENV key=value` and `ARG key=value`, where the variable name (`key`) and value (`value`) are
  separated by an equals sign (`=`). Historically, Dockerfiles have also supported a space
  separator between the key and the value (for example, `ARG key value`). This legacy format is
  deprecated, and you should only use the format with the equals sign.

This admonition does not mention `LABEL`, but it is included.

Update the use of `LABEL` and `ENV` where necessary to remove deprecation warning.

Fixes #486

### CODE REVIEW

File: amazoncorretto-8-windowsservercore/Dockerfile [hunk 1]
```diff
@@ -31,16 +31,16 @@ RUN Invoke-WebRequest -Uri ${env:BASE_URL}/apache-maven-${env:MAVEN_VERSION}-bin
...
+ENV MAVEN_CONFIG="$USER_HOME_DIR/.m2"
 
 COPY mvn-entrypoint.ps1 C:/ProgramData/Maven/mvn-entrypoint.ps1
 COPY settings-docker.xml C:/ProgramData/Maven/Reference/settings-docker.xml
 
 RUN setx /M PATH $('{0};{1}' -f $env:PATH,'C:\ProgramData\Maven\bin') | Out-Null
 
 USER ContainerUser
-ENV JAVA_HOME=${JAVA_HOME}
+ENV=JAVA_HOME=${JAVA_HOME}
```
Comment (carlossg) [inline amazoncorretto-8-windowsservercore/Dockerfile:43]:
```
Error response from daemon: dockerfile parse error on line 43: unknown instruction: ENV=JAVA_HOME=${JAVA_HOME}
```

Comment (timothystone-knsl) [reply amazoncorretto-8-windowsservercore/Dockerfile]: On it. Done. Pushed.

**Comment by carlossg** [comment]
FYI the `publish.sh` automates a bunch of things to avoid duplication. Changing the `Dockerfile-template` and running it updates all the other Dockerfiles

**Comment by timothystone** [comment]
I didn't see this feature in the `publish.sh` (well, I almost did, but I was looking at it for another reason, specifically if it opened the PR to the docker-library for release). I pulled your changes, but the `ENV=` returned in Amazon Corretto 8. Fixed, but if you want to rebase... I'll hold on pushing.

**Pull request merged**